### PR TITLE
[Merged by Bors] - zookeeper bundle for 23.4

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Bundle files for the 23.4 release of the zookeeper operator.

Test results when deployed from bundle:
```
--- PASS: kuttl (1829.93s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/logging_zookeeper-3.7.0-stackable23.4.0 (51.33s)
        --- PASS: kuttl/harness/znode_zookeeper-latest-3.8.0-stackable23.4.0 (27.94s)
        --- PASS: kuttl/harness/logging_zookeeper-3.6.3-stackable23.4.0 (66.48s)
        --- PASS: kuttl/harness/logging_zookeeper-3.5.8-stackable23.4.0 (60.61s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0_use-server-tls-true_use-client-auth-tls-true (80.02s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0_use-server-tls-true_use-client-auth-tls-false (72.00s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0_use-server-tls-false_use-client-auth-tls-true (68.31s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.8.0-stackable23.4.0_use-server-tls-false_use-client-auth-tls-false (57.74s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.5.8-stackable23.4.0_use-server-tls-true_use-client-auth-tls-false (78.94s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.5.8-stackable23.4.0_use-server-tls-false_use-client-auth-tls-true (71.87s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.5.8-stackable23.4.0_use-server-tls-false_use-client-auth-tls-false (57.92s)
        --- PASS: kuttl/harness/delete-rolegroup_zookeeper-3.8.0-stackable23.4.0 (55.97s)
        --- PASS: kuttl/harness/delete-rolegroup_zookeeper-3.7.0-stackable23.4.0 (66.71s)
        --- PASS: kuttl/harness/delete-rolegroup_zookeeper-3.6.3-stackable23.4.0 (55.81s)
        --- PASS: kuttl/harness/delete-rolegroup_zookeeper-3.5.8-stackable23.4.0 (213.87s)
        --- PASS: kuttl/harness/cluster-operation_zookeeper-latest-3.8.0-stackable23.4.0 (83.85s)
        --- PASS: kuttl/harness/logging_zookeeper-3.8.0-stackable23.4.0 (52.76s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.7.0-stackable23.4.0_use-server-tls-false_use-client-auth-tls-false (57.71s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.5.8-stackable23.4.0_use-server-tls-true_use-client-auth-tls-true (70.65s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.6.3-stackable23.4.0_use-server-tls-true_use-client-auth-tls-true (69.51s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.7.0-stackable23.4.0_use-server-tls-true_use-client-auth-tls-true (68.48s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.6.3-stackable23.4.0_use-server-tls-true_use-client-auth-tls-false (71.84s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.7.0-stackable23.4.0_use-server-tls-true_use-client-auth-tls-false (66.07s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.7.0-stackable23.4.0_use-server-tls-false_use-client-auth-tls-true (73.46s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.6.3-stackable23.4.0_use-server-tls-false_use-client-auth-tls-true (67.32s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.6.3-stackable23.4.0_use-server-tls-false_use-client-auth-tls-false (53.60s)
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] (Integration-)Test cases tested on Openshift (4.11)
```

```[tasklist]
# Reviewer
- [ ] Changelog updated
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: deploy/olm/23.4.0/manifests/configmap.yaml [hunk 1]
```diff
@@ -409,4 +409,4 @@ metadata:
   labels:
     app.kubernetes.io/name: zookeeper-operator
     app.kubernetes.io/instance: zookeeper-operator
-    app.kubernetes.io/version: "23.4.0-rc1"
+    app.kubernetes.io/version: "23.1.0"
```
Comment (razvan) [inline deploy/olm/23.4.0/manifests/configmap.yaml:412]:
```suggestion
    app.kubernetes.io/version: "23.4.0"
```

**Review by razvan — APPROVED** [review]
lgtm. One typo fix

File: deploy/olm/23.4.0/manifests/configmap.yaml (hunk 1 above)
Comment (adwk67) [reply deploy/olm/23.4.0/manifests/configmap.yaml:412]: Thanks for catching that!

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**